### PR TITLE
TRACING-4133| Make OpenTelemetry target allocator tech-preview

### PR DESCRIPTION
**Pull request by pavolloffay**
Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/TRACING-4133

Link to docs preview:

https://74417--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector#otel-config-target-allocator_otel-configuration-of-otel-collector

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by max-cx — APPROVED** [review]

File: modules/otel-config-target-allocator.adoc [hunk 1]
```diff
@@ -6,6 +6,8 @@
 [id="otel-config-target-allocator_{context}"]
 = Target allocator
 
+The Target allocator is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```
Comment (max-cx) [inline modules/otel-config-target-allocator.adoc:9]:
```suggestion
The target allocator is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```

Comment (pavolloffay) [reply modules/otel-config-target-allocator.adoc]: done

**Comment by tmalove** [comment]
@pavolloffay  Add the direct link to your update in the preview link of comment 0: 
https://74417--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector#otel-config-target-allocator_otel-configuration-of-otel-collector

/remove-label peer-review-in-progress
/label peer-review-done

**Review by adellape — COMMENTED** [review]
Hi, during merge-review, I identified some things we should address before merging:

* See in-line comment about using the Tech Preview snippet.
* I see that the TA feature is mentioned previously in the [3.1 release notes](https://74417--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel_rn/otel-rn-past-releases#otel-rn-3-1). That section should similarly be updated to note its TP status, or possibly reconfigured to appear under a "Technology Preview features" subheading, as I seen has been done for past release (e.g., [for 2.2](https://74417--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel_rn/otel-rn-past-releases#otel-rn_2-2_technology-preview-features)). (FWIW, depending on how OTEL release notes (RNs) are handled, it might be worth handling those in separate PRs per OCP branch version, versus trying to cherrypick this `main` PR all the way back to 4.12.)
* Per our openshift-docs CCS team processes, since we're changing the support status on something in published documentation, this PR will require what we call the "Change Management" process, gathering ACKs from relevant stakeholders (e.g., QE, Support, PM, etc.). I'll comment in the Jira issue with a link for more details, but I've also DM'd @max-cx as a heads-up as well.

And one out-of-scope thought:

* I am not sure about the relationship that may or not exist between the OTEL RNs and the core OCP RNs (seems like there is none really), but I just wanted to point out that in the OCP RNs there are Tech Preview tracker tables, e.g. [here](https://docs.openshift.com/container-platform/4.15/release_notes/ocp-4-15-release-notes.html#ocp-4-15-technology-preview-tables). No action required currently really, but for the future could consider something similar in OTEL RNs, or maybe add a general "Observability Technology Preview features" tracker in the OCP RNs for extra visibility. 

I will leave the `merge-review-in-progess` label on it so that it remains in my views even after I'm done with merge-review squad this week, but please reach back out when it is ready for re-review and when CM ACKs are in place. Thank you!

File: modules/otel-config-target-allocator.adoc [hunk 2]
```diff
@@ -6,6 +6,8 @@
 [id="otel-config-target-allocator_{context}"]
 = Target allocator
 
+The target allocator is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-target-allocator.adoc:9]: @adellape, thank you for this feedback.
 
@pavolloffay, I'm working on an edit.

[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-config-target-allocator.adoc]: @pavolloffay, regarding the release notes fix, could you please cherry-pick my commit from https://github.com/openshift/openshift-docs/pull/74801?

**Comment by max-cx** [comment]
@adellape, thank you for your assistance with this PR. I have followed the Change Management process steps, and the before-change [Notice of Change] email sent earlier this week. Do you think you could merge this PR?

**Review by adellape — APPROVED** [review]

**Pull request merged**